### PR TITLE
Implement KillIngressSession rpc

### PR DESCRIPTION
**Pull request by biglittlebigben**
Also remove unused H264 depacketization code and return resourceIDs in ListActiveIngress.

### CODE REVIEW

File: pkg/service/process_manager.go [hunk 1]
```diff
@@ -77,8 +78,27 @@ func (s *ProcessManager) startIngress(ctx context.Context, p *params.Params, clo
...
+
+			go func() {
+				time.Sleep(time.Second)
+
+				s.mu.Lock()
+				h := s.activeHandlers[p.State.ResourceId]
+
+				if h != nil && !h.closed.IsBroken() && h.cmd != nil {
+					logger.Infow("killing handler process still present after termination was requested", "ingressID", h.info.IngressId, "resourceID", p.State.ResourceId, "startedAt", p.State.StartedAt)
+					if err := h.cmd.Process.Signal(syscall.SIGKILL); err != nil {
```
Comment (dennwc) [inline pkg/service/process_manager.go:95]: Not sure if it's important, but `killAll` uses `SIGINT`, while this line uses `SIGKILL`. Probably both using `SIGKILL` in both?

File: pkg/service/process_manager.go [hunk 2]
```diff
@@ -77,8 +78,27 @@ func (s *ProcessManager) startIngress(ctx context.Context, p *params.Params, clo
...
+		sessionCloser: func(ctx context.Context) {
+			if closeSession != nil {
+				closeSession(ctx)
+			}
+
+			go func() {
+				time.Sleep(time.Second)
+
+				s.mu.Lock()
+				h := s.activeHandlers[p.State.ResourceId]
```
Comment (dennwc) [inline pkg/service/process_manager.go:91]: Is it possible to unlock as soon as `h` is returned from the map?

File: pkg/service/process_manager.go [hunk 3]
```diff
@@ -149,6 +169,9 @@ func (s *ProcessManager) runHandler(ctx context.Context, h *process, p *params.P
 		case errors.As(err, &exitErr):
 			if exitErr.ProcessState.ExitCode() == 1 {
 				logger.Infow("relaunching handler process after retryable failure")
+			} else if err.Error() == "signal: killed" {
```
Comment (dennwc) [inline pkg/service/process_manager.go:172]: Too bad that `exitErr` doesn't have any details except the text.

File: pkg/service/session_manager.go [hunk 4]
```diff
@@ -64,23 +67,25 @@ func (sm *SessionManager) IngressStarted(info *livekit.IngressInfo, sessionAPI t
...
 
-func (sm *SessionManager) IngressEnded(info *livekit.IngressInfo) {
-	logger.Infow("ingress ended", "ingressID", info.IngressId, "resourceID", info.State.ResourceId)
-
+func (sm *SessionManager) IngressEnded(resourceID string) {
 	sm.lock.Lock()
 	defer sm.lock.Unlock()
 
-	p := sm.sessions[info.State.ResourceId]
+	p := sm.sessions[resourceID]
```
Comment (dennwc) [inline pkg/service/session_manager.go:79]: If I understood correctly, `ListIngress` may return an empty `resourceID`, which will be skipped here. Not sure how frequent that is. Is it worth checking before locking the mutex?

**Review by dennwc — APPROVED** [review]

File: pkg/service/process_manager.go (hunk 1 above)
Comment (dennwc) [reply pkg/service/process_manager.go:95]: Thank you for the explanation, make sense now! Was confused by "kill" part in `killAll`.

File: pkg/service/process_manager.go (hunk 2 above)
Comment (biglittlebigben) [reply pkg/service/process_manager.go:91]: Sure

File: pkg/service/session_manager.go (hunk 4 above)
Comment (biglittlebigben) [reply pkg/service/session_manager.go:79]: We need to deploy ingress before cloud-io. Nobody will call KillIIngressSession before we do so.

**Pull request merged**